presto-spi/src/main/java/io/prestosql/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -481,6 +562,172 @@ public static void multiply(Slice left, Slice right, Slice result)
         pack(result, (int) z0, (int) z1, (int) z2, (int) z3, leftNegative != rightNegative);
     }
 
+    public static void multiply(Slice left, long right, Slice result)
+    {
+        checkArgument(result.length() == NUMBER_OF_LONGS * Long.BYTES);
+
+        long l0 = toUnsignedLong(getInt(left, 0));
```

Comment:
copy and paste, somebody fixes an issue in one place, forgets another and we have a bug; generalize the methods, or if not possible (eg would hurt performance), write a script which generates this copy and paste